### PR TITLE
fix vector handling and broadcasting in multivariate_normal density

### LINKED ISSUE #626: multivariate normal density cannot handle vectors as it claims
It seems that if `x` and `mu` are vectors, the call to `tf.matrix_triangular_solve` would fail (as `d = x - mu` is rank 1). Worse, if `x` and `mu` are of different dimensionality (e.g. one a row, one a column vector) broadcasting happens and fails silently.

### PR DESCRIPTION
**Pull request by ialong**
Fixes #626.

### CODE REVIEW

File: gpflow/densities.py [hunk 1]
```diff
@@ -77,11 +77,11 @@ def multivariate_normal(x, mu, L):
     assume independence over the *columns*: the number of rows must match the
     size of L.
     """
-    d = x - mu
+    d = tf.cond(tf.rank(x) < 2, lambda: x[:, None], lambda: x) \
```
Comment (awav) [inline gpflow/densities.py]: Add some readability to the code:
```
x = tf.cond(tf.rank(x) < 2, lambda: x[:, None], lambda: x)
mu = tf.cond(tf.rank(mu) < 2, lambda: mu[:, None], lambda: mu)
d = x - mu
```
and num_col and num_dims types depend on L type
```
num_col = tf.cast(tf.shape(d)[1], L.dtype)
num_dims = tf.cast(tf.shape(d)[0], L.dtype)
```

Comment (ialong) [reply gpflow/densities.py]: Sounds good! (the use of settings.float_type was already there but your change sounds sensible)

**Comment by awav** [comment]
It would be nice to have some tests covering 4 cases, *all rank 2*, *mu rank 2*, *x rank 2*, *none rank 2* :)

**Comment by ialong** [comment]
@awav added the tests

**Pull request merged**